SpecNuts/ReportingAspect/ReportingMessageSink.cs

```diff
@@ -19,26 +20,40 @@ public IMessageCtrl AsyncProcessMessage(IMessage msg, IMessageSink replySink)
 
 		public IMessage SyncProcessMessage(IMessage msg)
 		{
-			var methodMessage = new MethodCallMessageWrapper((IMethodCallMessage) msg);
+			var methodMessage = new MethodCallMessageWrapper((IMethodCallMessage)msg);
 
-			IMethodReturnMessage mrm = null;
+			// Avoid reporting proxy method calls for fields injected via the constructor
+			// in the step definition classes. This caused some steps to pe reported twice.
+			if (methodMessage.MethodName == "FieldGetter")
+			{
+				return NextSink.SyncProcessMessage(msg);
+			}
 
-			Reporters.ExecuteStep(
+			IMethodReturnMessage mrm = null;
+			var task = Reporters.ExecuteStep(
```

Comment:
The fix consists in wrapping the step definition task with a task that allows reporting the execution of that step.

Please note that if the return type was not a task, we're passing to the ExecuteStep method a completed Task, so it doesn't impact the step definitions that return void.